sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -198,9 +199,15 @@ public void SetProperty(string name, object value)
         /// </summary>
         public void Dispose()
         {
-            Request?.Dispose();
-            _response?.Dispose();
+            Request.Dispose();
```

Comment:
Don't we need a similar check to avoid double-disposing Request? Why not just use a _disposed flag?

Reply:
Ah I guess the double-dispose fix is in ArrayBackedPropertyBag itself.